src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -147,6 +156,102 @@ public void handleHelp() {
+        if (!Desktop.isDesktopSupported()) {
+            messageWindow.show("Desktop not supported.");
+            return;
+        }
+
+        logger.info("Loading from csv");
+        Path inputCsvFilePath = handleLoadFile();
+
+        if (inputCsvFilePath == null) {
+            return;
+        }
+        boolean isSuccess = logic.loadAddressBookFromCsv(inputCsvFilePath);
+        if (isSuccess) {
+            messageWindow.show("Loaded successfully.");
+        } else {
+            messageWindow.show("Error in loading.");
+        }
+    }
+
+    /**
+     * OHandles the tasks associated with allowing the user to save a file to CSV.
+     */
+    @FXML
+    public void handleSaveToCsv() {
+        if (!Desktop.isDesktopSupported()) {
+            messageWindow.show("Desktop not supported.");
+            return;
+        }
+
+        logger.info("Saving to csv");
+        Path outputCsvFilePath = handleSaveFile();
+
+        if (outputCsvFilePath == null) {
+            return;
+        }
+        boolean isSuccess = logic.saveAddressBookToCsv(outputCsvFilePath);
+        if (isSuccess) {
+            messageWindow.show("Saved successfully.");
+        } else {
+            messageWindow.show("Error in saving.");
+        }
+    }
+
+    public Path handleLoadFile() {
+
+        // initialise the file chooser
+        JFileChooser chooser = new JFileChooser();
+
+        // current working directory
+        File cwd = new java.io.File(".");
+
+        // chooser settings
+        chooser.setCurrentDirectory(cwd);
+        chooser.setDialogTitle("Select a ClientConnect CSV file...");
+        chooser.setAcceptAllFileFilterUsed(false);  // disable allowing acceptance of all files
+
+        // only allow CSV files
+        FileNameExtensionFilter filter = new FileNameExtensionFilter("ClientConnect CSV files", "csv");
+        chooser.setFileFilter(filter);
+
+        // opens the dialog
+        int returnVal = chooser.showOpenDialog(null);
+        Path chosenFilePath = null;
+        if (returnVal == JFileChooser.APPROVE_OPTION) {
+            chosenFilePath = chooser.getSelectedFile().toPath().toAbsolutePath();
+        }
+        return chosenFilePath;
+
+    }
+
+    public Path handleSaveFile() {
+
+        // initialise the file chooser
+        JFileChooser chooser = new JFileChooser();
+
+        chooser.setSelectedFile(new File("ClientConnectData.csv"));
+
+        // chooser settings
+        chooser.setCurrentDirectory(new java.io.File("."));
+        chooser.setDialogTitle("Save CSV file to ...");
+        chooser.setSelectedFile(new File("ClientConnectData.csv"));
+
+        // opens the dialog
+        int returnVal = chooser.showSaveDialog(null);
+        Path chosenDirectoryPath = null;
+        if (returnVal == JFileChooser.APPROVE_OPTION) {
+            chosenDirectoryPath = chooser.getSelectedFile().toPath().toAbsolutePath();
+        }
+        return chosenDirectoryPath;
+    }
```

Comment:
Could these error messages be stored in external variables to reduce the duplication of code and make it more understandable?